### PR TITLE
feat(radixmenu): add support for data-testid on RadixMenu + add dark variant + other adventures

### PR DESCRIPTION
**Pull request by rnmp**

### CODE REVIEW

**Review by Crockery — APPROVED** [review]
Looks great!

File: packages/zephyr/src/RadixMenu/components/RadixMenuItemDivider.tsx [hunk 1]
```diff
@@ -2,16 +2,24 @@ import React, { memo, FC, ComponentProps } from 'react';
...
 }
 
-export const RadixMenuItemDivider = memo(({ isTop }: RadixMenuItemDividerProps) => {
+export const RadixMenuItemDivider = memo(({ isTop, variant }: RadixMenuItemDividerProps) => {
   return (
     <Box
       as={Separator as FC<Omit<ComponentProps<typeof Separator>, 'as' | 'ref'>>}
-      tx={{ backgroundColor: 'pigeon100', height: 1, border: 0, mt: isTop ? 0 : 8, mb: 8 }}
+      tx={{
+        backgroundColor: variant === 'dark' ? 'pigeon700' : 'pigeon100',
```
Comment (andyhqtran) [inline packages/zephyr/src/RadixMenu/components/RadixMenuItemDivider.tsx:17]:
```suggestion
        backgroundColor: variant === 'dark' ? 'pigeon500' : 'pigeon100',
```

File: packages/zephyr/src/RadixMenu/components/RadixMenuItem.tsx [hunk 2]
```diff
@@ -131,7 +142,10 @@ export const RadixMenuItem = memo(
                 <MenuItemLabel variant={isSmallSize ? 'text-ui-14' : 'text-ui-16'}>
                   {renderProps.label}
                 </MenuItemLabel>
-                <MenuItemDescription variant={isSmallSize ? 'text-ui-12' : 'text-ui-14'}>
+                <MenuItemDescription
+                  variant={isSmallSize ? 'text-ui-12' : 'text-ui-14'}
+                  tx={variant === 'dark' ? { color: 'pigeon400' } : undefined}
```
Comment (andyhqtran) [inline packages/zephyr/src/RadixMenu/components/RadixMenuItem.tsx:147]:
```suggestion
                  tx={variant === 'dark' ? { color: 'pigeon100' } : undefined}
```

File: packages/zephyr/src/RadixMenu/components/RadixMenuItem.tsx [hunk 3]
```diff
@@ -97,21 +102,27 @@ export const RadixMenuItem = memo(
           py: hasDescription ? 6 : 0,
           border: 0,
           borderRadius: 4,
-          color: 'pigeon700',
+          color: variant === 'dark' ? 'pigeon200' : 'pigeon700',
```
Comment (andyhqtran) [inline packages/zephyr/src/RadixMenu/components/RadixMenuItem.tsx:105]:
```suggestion
          color: variant === 'dark' ? 'pigeon100' : 'pigeon700',
```

File: packages/zephyr/src/RadixMenu/components/RadixMenuItem.tsx [hunk 4]
```diff
@@ -97,21 +102,27 @@ export const RadixMenuItem = memo(
...
           cursor: 'pointer',
 
-          '&[data-selected]': { backgroundColor: 'pigeon050', color: 'pigeon700' },
+          '&[data-selected], &:active': {
+            backgroundColor: variant === 'dark' ? 'pigeon800' : 'pigeon050',
+            color: variant === 'dark' ? 'pigeon100' : 'pigeon700',
+          },
 
-          '&:hover': { backgroundColor: 'pigeon050' },
+          '&:hover': { backgroundColor: variant === 'dark' ? 'pigeon700' : 'pigeon050' },
```
Comment (andyhqtran) [inline packages/zephyr/src/RadixMenu/components/RadixMenuItem.tsx:114]:
```suggestion
          '&:hover': {
            backgroundColor: variant === 'dark' ? 'pigeon800' : 'pigeon050',
            color: variant === 'dark' ? 'white' : undefined,
          },
```

**Review by andyhqtran — COMMENTED** [review]
Here are the mocks that we have for dark menu, the colors seems a bit off + hard to read
https://www.figma.com/file/tZC7d1yYElZlqB8EmIQ0zT/Multiselect-bar-02.21?node-id=1770%3A0

**Review by andyhqtran — APPROVED** [review]

**Pull request merged**